### PR TITLE
Fixed:- retired courses certificates for profile & learning strip in home page

### PR DESCRIPTION
**Pull request by Afzal84**
Fixed:- retired courses certificates for profile & learning strip in home page

### CODE REVIEW

File: library/ws-widget/collection/src/lib/card-content-v2/card-content-v2.component.html [hunk 1]
```diff
@@ -1099,8 +1099,8 @@ <h2 class="mat-subheading-2 flex-1 min-w-0 margin-remove-bottom" i18n>{{'cardcon
 </ng-template>
 
 <ng-template #cardPortrait>
-    <mat-card class="card-portrait cursor-pointer" [ngClass]="widgetData.cardCustomeClass ? widgetData.cardCustomeClass : ''" (click)="raiseTelemetry();getRedirectUrlData(widgetData?.content); $event.stopPropagation()" >
-      <div class="course_widget portrait_widget flex  margin-bottom-xs">
+    <mat-card class="card-portrait cursor-pointer" [ngClass]="widgetData.cardCustomeClass ? widgetData.cardCustomeClass : ''"  (click)="raiseTelemetry();getRedirectUrlData(widgetData?.content); $event.stopPropagation()" >
+      <div class="course_widget portrait_widget flex  margin-bottom-xs" [ngClass]="{'retiredCards': widgetData.content?.status === 'Retired' && widgetData.content?.completionPercentage === 100}">
```
Comment (vishnubansaltarento) [inline library/ws-widget/collection/src/lib/card-content-v2/card-content-v2.component.html]: please put ? against widgetData as well

File: library/ws-widget/collection/src/lib/card-content-v2/card-content-v2.component.html [hunk 2]
```diff
@@ -1110,7 +1110,7 @@ <h2 class="mat-subheading-2 flex-1 min-w-0 margin-remove-bottom" i18n>{{'cardcon
           </ws-widget-display-content-type>
         </div>
       </div>
-      <div class="flex">
+      <div class="flex" [ngClass]="{'retiredCards': widgetData.content?.status === 'Retired' && widgetData.content?.completionPercentage === 100}">
```
Comment (vishnubansaltarento) [inline library/ws-widget/collection/src/lib/card-content-v2/card-content-v2.component.html]: same as above

File: project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts [hunk 3]
```diff
@@ -157,7 +157,29 @@ export class SeeAllHomeComponent implements OnInit, OnDestroy {
...
+        }
+      } else {
+        completed.push(e)
+      }
+    })
+    // Sort the completed array with 'live' status first and 'Retired' status second
+    completed.sort((a: any, b: any) => {
+      if (a.status === 'live' && b.status !== 'live') {
+        return 1
+      }  if (a.status !== 'live' && b.status === 'live') {
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts]: please check the status by converting it to lowercase

Comment (Afzal84) [reply project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts]: converted plz check

File: library/ws-widget/collection/src/lib/card-content-v2/card-content-v2.component.html (hunk 2 above)
Comment (Afzal84) [reply library/ws-widget/collection/src/lib/card-content-v2/card-content-v2.component.html]: changed

File: library/ws-widget/collection/src/lib/card-content-v2/card-content-v2.component.html (hunk 1 above)
Comment (Afzal84) [reply library/ws-widget/collection/src/lib/card-content-v2/card-content-v2.component.html]: updated

File: library/ws-widget/collection/src/lib/content-strip-with-tabs/content-strip-with-tabs.component.ts [hunk 4]
```diff
@@ -445,7 +451,35 @@ export class ContentStripWithTabsComponent extends WidgetBaseComponent
                             strip: NsContentStripWithTabs.IContentStripUnit) {
     const inprogress: any[] = []
     const completed: any[] = []
-    array.forEach((e: any, idx: number, arr: any[]) => (customFilter(e, idx, arr) ? inprogress : completed).push(e))
+    // array.forEach((e: any, idx: number, arr: any[]) => (customFilter(e, idx, arr) ? inprogress : completed).push(e))
+    array.forEach((e, idx, arr) => {
+      const statusRetired = (e.status as string).toLowerCase()
```
Comment (vishnubansaltarento) [inline library/ws-widget/collection/src/lib/content-strip-with-tabs/content-strip-with-tabs.component.ts]: first put condition for e.status , then do lowercase conversion

File: library/ws-widget/collection/src/lib/content-strip-with-tabs/content-strip-with-tabs.component.ts [hunk 5]
```diff
@@ -445,7 +451,35 @@ export class ContentStripWithTabsComponent extends WidgetBaseComponent
...
+        if (statusRetired !== 'retired') {
+          inprogress.push(e)
+        }
+      } else {
+        completed.push(e)
+      }
+    })
+    // Sort the completed array with 'Live' status first and 'Retired' status second
+    completed.sort((a: any, b: any) => {
+      const statusA = a.status.toLowerCase()
```
Comment (vishnubansaltarento) [inline library/ws-widget/collection/src/lib/content-strip-with-tabs/content-strip-with-tabs.component.ts]: same here

File: project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts [hunk 6]
```diff
@@ -176,7 +176,34 @@ export class SeeAllHomeComponent implements OnInit, OnDestroy {
...
+        if (statusRetired !== 'retired') {
+          inprogress.push(e)
+        }
+      } else {
+        completed.push(e)
+      }
+    })
+    // Sort the completed array with 'live' status first and 'Retired' status second
+    completed.sort((a: any, b: any) => {
+      const statusA = a.status.toLowerCase()
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts]: same as above

File: project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts [hunk 7]
```diff
@@ -176,7 +176,34 @@ export class SeeAllHomeComponent implements OnInit, OnDestroy {
                             strip: NsContentStripWithTabs.IContentStripUnit) {
     const inprogress: any[] = []
     const completed: any[] = []
-    array.forEach((e: any, idx: number, arr: any[]) => (customFilter(e, idx, arr) ? inprogress : completed).push(e))
+    array.forEach((e, idx, arr) => {
+      const statusRetired = (e.status as string).toLowerCase()
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts]: same as above

Comment (Afzal84) [reply project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts]: done

File: project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts (hunk 6 above)
Comment (Afzal84) [reply project/ws/app/src/lib/routes/see-all/components/see-all-home/see-all-home.component.ts]: changed

File: library/ws-widget/collection/src/lib/content-strip-with-tabs/content-strip-with-tabs.component.ts (hunk 5 above)
Comment (Afzal84) [reply library/ws-widget/collection/src/lib/content-strip-with-tabs/content-strip-with-tabs.component.ts]: channged

File: library/ws-widget/collection/src/lib/content-strip-with-tabs/content-strip-with-tabs.component.ts (hunk 4 above)
Comment (Afzal84) [reply library/ws-widget/collection/src/lib/content-strip-with-tabs/content-strip-with-tabs.component.ts]: changed

**Pull request merged**